File: scripts/crop_chelsa.R
# snakemake log
log_file <- file(snakemake@log[[1]], open = "wt")
sink(log_file, append = TRUE, type = "message")
sink(log_file, append = TRUE)

# snakemake vars
filein <- snakemake@input[[1]]
fileout <- snakemake@output[[1]]
foldergadm <- snakemake@input[[2]]
country <- snakemake@params$country

# test
# country <- "New-Caledonia"
# filein <- "results/chelsa/raw/pr_01_1979.tif"
# folderout <- "results/countries/New-Caledonia_gadm/"

# libraries
library(tidyverse)
library(geodata)
library(sf)
library(terra)

# code
country <- gsub("-", " ", country)
bb <- gadm(country, level = 0, path = foldergadm) %>% 
  st_as_sf() %>% 
  st_union() %>% 
  st_bbox() %>% 
  st_as_sfc()
rast(filein) %>% 
  crop(bb) %>% 
  writeRaster(fileout)

File: rules/crop_chelsa.py
rule crop_chelsa:
    input:
        "results/chelsa/raw/{var}_{month}_{year}.tif",
        "results/countries/{country}_gadm"
    output:
        temp("results/chelsa/cropped/{var}_{country}_{month}_{year}.tif")
    log:
        "results/logs/crop_chelsa_{var}_{country}_{month}_{year}.log"
    benchmark:
        "results/benchmarks/crop_chelsa_{var}_{country}_{month}_{year}.benchmark.txt"
    threads: 1
    resources:
        mem_mb=1000
    params:
        country="{country}"
    script:
      "../scripts/crop_chelsa.R"
